keanu-project/src/test/java/io/improbable/keanu/util/dot/WriteDotTest.java

```diff
@@ -0,0 +1,166 @@
+package io.improbable.keanu.util.dot;
+
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.BooleanIfVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.ConstantBoolVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary.MultiplicationVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.ConstantIntegerVertex;
+import org.apache.commons.io.FileUtils;
+import org.junit.*;
+import org.nd4j.util.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+public class WriteDotTest {
+
+    private static BayesianNetwork gaussianNet;
+    private static BayesianNetwork booleanNet;
+    private static String testFileDir = "dottestdir";
+    private static String gaussianDotOutput;
+    private static String boolDotOutput;
+    private static String vertexDegree1DotOutput;
+    private static String vertexDegree2DotOutput;
+    private static String boolVertexLabel = "/BoolResult";
+    private static GaussianVertex gaussianV;
+    private static DoubleVertex doubleV;
+
+    @BeforeClass
+    public static void setup() throws IOException {
+
+        String gaussianDotFileName = testFileDir + "/gaussianDot.dot";
+        String boolDotFileName = testFileDir + "/boolDot.dot";
+        String vertexDegree1DotFileName = testFileDir + "/VertexOutput1.dot";
+        String vertexDegree2DotFileName = testFileDir + "/VertexOutput2.dot";
+
+        double mu = 0;
+        double sigma = 1;
+        gaussianV = new GaussianVertex(mu, sigma);
+        doubleV = new ConstantDoubleVertex(5.43);
+        DoubleVertex result = gaussianV.multiply(doubleV);
+        gaussianNet = new BayesianNetwork(result.getConnectedGraph());
+        WriteDot.outputDot(gaussianDotFileName, gaussianNet);
+        gaussianDotOutput = new String(Files.readAllBytes(Paths.get(gaussianDotFileName)), Charset.defaultCharset());
+
+
+        ConstantBoolVertex boolV = new ConstantBoolVertex(true);
```

Comment:
I'd go for some more descriptive names here. Like trueConstant...

Reply:
Oki, changed.